corehq/project_limits/rate_limiter.py

```diff
@@ -50,6 +50,13 @@ def report_usage(self, scope=None, delta=1):
         for rate_counter, limit in self.get_rate_limits(*scope):
             rate_counter.increment((self.feature_key,) + scope, delta=delta)
 
+    def rate_limit_exceeded_for_window(self, scope=None):
```

Comment:
I don't like that I created a new method for this, since `allow_usage` does the same thing. I needed to change the functionality to return the time window that was triggered for the rate limit, but I saw other places using `allow_usage`. They could be updated to handle a different return value, but appear to do fine with a bool value. 

I think my best option is to update `allow_usage` to pass through to `rate_limit_exceeded_for_window` and return a bool value depending on the string returned. That way the caller is still isolated from the change.

Reply:
Yeah I see what you mean. I know this isn't what you were asking, but I'd change the name of this method to something like `get_window_of_first_exceeded_limit`. (As is it sounds like you're supposed to pass in a window to check.)

It's a little painful how similar the methods are, and tempting to try to jam them into one function that returns both, but anything I can think of is a tad too clever. So I think your plan (basically making allow_usage an alias for `get_window_of_first_exceeded_limit(...) is None`) is the least surprising and thus the best.

Reply:
That or leaving `allow_usage` as-is would be fine too I think. Most of the logic is contained in `iter_rates`, so what duplication there is isn't that offensive to me

Reply:
Much better method name! Thanks. I'll leave `allow_usage` as is.